prov/kubernetes/executor.go

```diff
@@ -55,14 +57,27 @@ func (e *defaultExecutor) ExecAsyncOperation(ctx context.Context, conf config.Co
 		return nil, 0, err
 	}
 
-	if isNilValue(e.clientset) {
-		e.clientset, err = initClientSet(conf)
-		if err != nil {
-			return nil, 0, err
-		}
+	var locationProps config.DynamicMap
+	locationMgr, err := locations.NewManager(conf)
```

Comment:
Is there a good reason for recreating the manager for each execution? Does it worth caching it?